testsrc/main/mondrian/rolap/NativeFilterAgainstAggTableTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+// This software is subject to the terms of the Eclipse Public License v1.0
+// Agreement, available at the following URL:
+// http://www.eclipse.org/legal/epl-v10.html.
+// You must accept the terms of that agreement to use this software.
+//
+// Copyright (c) 2015-2015 Pentaho Corporation.  All rights reserved.
+*/
+package mondrian.rolap;
+
+import mondrian.test.SqlPattern;
+
+/**
+ * @author Andrey Khayrutdinov
+ */
+public class NativeFilterAgainstAggTableTest extends BatchTestCase {
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        propSaver.set(propSaver.properties.UseAggregates, true);
+        propSaver.set(propSaver.properties.ReadAggregates, true);
+        propSaver.set(propSaver.properties.EnableNativeFilter, true);
+    }
+
+    public void testFilteringOnAggregated_ByCount() {
+        // http://jira.pentaho.com/browse/MONDRIAN-2155
+        // Aggregation table can have fact's count value exceeding 1,
+        // so that to compute the overall amount of facts it is necessary
+        // to sum the values instead of counting them
+
+        // See this query:
+        //      select
+        //          count(t.fact_count) as cnt,
+        //          sum(t.fact_count) as s
+        //      from
+        //          agg_c_14_sales_fact_1997 as t
+        //      join
+        //          product as p on
+        //              t.product_id = p.product_id and
+        //              t.the_year=1997 and t.quarter='Q1'
+        //      join
+        //          product_class as pp on
+        //              pp.product_class_id = p.product_class_id and
+        //              pp.product_family = 'Food'
+        // It returns:
+        //      +-------+-------+
+        //      | cnt   | s     |
+        //      +-------+-------+
+        //      | 15533 | 15539 |
+        //      +-------+-------+
+
+        String query = ""
+            + "SELECT "
+            + "   {FILTER("
+            + "      {[Product].[All Products].Children},"
+            + "      [Measures].[Sales Count] < 15535"
+            + "   )} ON COLUMNS,"
+            + "   {[Measures].[Sales Count]} on ROWS "
+            + "FROM [Sales] "
+            + "WHERE [Time].[1997].[Q1]";
+
+        String expectedResult = ""
+            + "Axis #0:\n"
+            + "{[Time].[1997].[Q1]}\n"
+            + "Axis #1:\n"
+            + "{[Product].[Drink]}\n"
+            + "{[Product].[Non-Consumable]}\n"
+            + "Axis #2:\n"
+            + "{[Measures].[Sales Count]}\n"
+            + "Row #0: 1,959\n"
+            + "Row #0: 4,090\n";
+
+        doTestFilteringOnAggregatedBy("COUNT", query, expectedResult);
+    }
+
+    public void testFilteringOnAggregated_ByDistinctCount() {
+        String query = ""
```

Comment:
This test does not actually use an aggregate table for the native SqlTupleReader query.